cmd/main.go

```diff
@@ -0,0 +1,116 @@
+// Package cmd implements the commands supported by opfcli.
+package cmd
+
+import (
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/operate-first/opfcli/utils"
+)
+
+const defaultAppName = "cluster-scope"
+const namespacePath = "base/core/namespaces"
+const groupPath = "base/user.openshift.io/groups"
+const componentPath = "components"
+const componentRelPath = "../../../../components"
+
+var config = viper.New()
+
+var cfgFile string
+var appName string
+var repoDirectory string
+
+var rootCmd = &cobra.Command{
+	Use:   "opfcli",
+	Short: "A command line tool for Operate First GitOps",
+	Long: `A command line tool for Operate First GitOps.
+
+Use opfcli to interact with an Operate First style Kubernetes
+configuration repository.`,
+}
+
+// Execute processes the opfcli command line.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		log.Fatalf("Error: %s", err)
+	}
+}
+
+func init() {
+	loglevel, err := strconv.Atoi(os.Getenv("OPF_LOGLEVEL"))
+	if err != nil {
+		loglevel = 1
+	}
+
+	switch {
+	case loglevel >= 2:
+		log.SetLevel(log.DebugLevel)
+	case loglevel >= 1:
+		log.SetLevel(log.InfoLevel)
+	default:
+		log.SetLevel(log.WarnLevel)
+	}
+
+	cobra.OnInitialize(initConfig)
+
+	rootCmd.PersistentFlags().StringVarP(
+		&cfgFile, "config", "f", "", "configuration file")
+	rootCmd.PersistentFlags().StringVarP(
+		&appName, "app-name", "a", "cluster-scope", "application name")
+	rootCmd.PersistentFlags().StringVarP(
+		&repoDirectory, "repodir", "R", "", "path to opf repository")
+
+	if err = config.BindPFlag("app-name", rootCmd.PersistentFlags().Lookup("app-name")); err != nil {
+		panic(err)
```

Comment:
In the other parts of the code it seems the program exits via `log.Fatalf` any reason why we're doing a panic here instead of reporting a fatal error and exiting? (I'm not too versed on when we ought to panic over os.exit etc.)

Reply:
So, my though here was that this happens before logging is configured, and honestly this should *never* fail. Checking the error here is really just to make linters happy. Because a problem here represents a "how could this possible fail" situation, I figured a `panic` as appropriate.

Reply:
Ah that makes sense to me, thanks :+1: